model/jpa/src/main/java/org/keycloak/models/jpa/JpaUserProvider.java

```diff
@@ -841,42 +841,60 @@ public int getUsersCount(Map<String, String> params, RealmModel realm, Set<Strin
             predicates.add(root.get("serviceAccountClientLink").isNull());
         }
 
-        for (Map.Entry<String, String> entry : attributes.entrySet()) {
-            String key = entry.getKey();
-            String value = entry.getValue();
+        final String idpAlias = attributes.get(UserModel.IDP_ALIAS);
+        final String idpUserId = attributes.get(UserModel.IDP_USER_ID);
+        if (idpAlias != null) {
```

Comment:
This introduction of `if-then-else` introduces a separate branch of the search. Is there a reason for not treating `UserModel.IDP_ALIAS` and `UserModel.IDP_USER_ID` the same way as the other attributes? Could it be part of the `switch` instead?

Reply:
My colleagues and I thought that it would be better to have the idp-search separated from the search for other attributes. One reason was that we decided that the idp-parameters have to be always searched for exactly - this is different from the other attributes for which a contains-search is provided.

But I can also understand the advantage of combining the idp-search with other attributes - the search would be more powerful.

Reply:
> One reason was that we decided that the idp-parameters have to be always searched for exactly - this is different from the other attributes for which a contains-search is provided.

This limitation is a reasonable one, since it would be strange to search for an identity provider alias / username via a substring search. Still, JPA storage search would benefit from allowing for combining the predicates, so that it would be possible to search for e.g. enabled users coming from the given idp.